subprojects/core/src/main/java/org/gradle/internal/buildevents/BuildExceptionReporter.java

```diff
@@ -129,58 +132,82 @@ private void reportBuildFailure(String granularity, Throwable failure, FailureDe
 
         fillInFailureResolution(details);
 
-        if (failure instanceof LocationAwareException) {
-            final LocationAwareException scriptException = (LocationAwareException) failure;
+        if (isInternalFailure(failure)) {
+            reportInternalGradleError(details, failure);
+        } else if (failure instanceof LocationAwareException) {
+            LocationAwareException scriptException = (LocationAwareException) failure;
             details.failure = scriptException.getCause();
             if (scriptException.getLocation() != null) {
                 details.location.text(scriptException.getLocation());
             }
-            scriptException.visitReportableCauses(new TreeVisitor<Throwable>() {
-                int depth;
-
-                @Override
-                public void node(final Throwable node) {
-                    if (node == scriptException) {
-                        details.details.text(getMessage(scriptException.getCause()));
-                    } else {
-                        final LinePrefixingStyledTextOutput output = getLinePrefixingStyledTextOutput();
-                        output.text(getMessage(node));
-                    }
-                }
-
-                @Override
-                public void startChildren() {
-                    depth++;
-                }
-
-                @Override
-                public void endChildren() {
-                    depth--;
-                }
-
-                private LinePrefixingStyledTextOutput getLinePrefixingStyledTextOutput() {
-                    details.details.format("%n");
-                    StringBuilder prefix = new StringBuilder();
-                    for (int i = 1; i < depth; i++) {
-                        prefix.append("   ");
-                    }
-                    details.details.text(prefix);
-                    prefix.append("  ");
-                    details.details.style(Info).text("> ").style(Normal);
-
-                    return new LinePrefixingStyledTextOutput(details.details, prefix, false);
-                }
-            });
+            details.details.text(getMessage(details.failure));
+            scriptException.visitReportableCauses(new ExceptionFormattingVisitor(scriptException, details));
         } else {
             details.details.text(getMessage(failure));
         }
     }
 
+    private static boolean isInternalFailure(Throwable failure) {
+        if (failure instanceof LocationAwareException) {
+            failure = failure.getCause();
+        }
+        return failure instanceof ServiceCreationException || failure instanceof NativeException;
+    }
+
+    private static void reportInternalGradleError(FailureDetails details, Throwable failure) {
+        details.details.text("Gradle could not start your build.");
+        LinePrefixingStyledTextOutput output = getLinePrefixingStyledTextOutput(details, 0);
+        output.text(Throwables.getRootCause(failure).getMessage());
```

Comment:
I think this looks similar to what we're doing with getReportableCauses in LocationAwareException, only we drill down into the ultimate cause.

Could we attach `@Contextual` to ServiceCreationException so we report the lower level exceptions vs special casing internal exceptions here?